platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentActivationOptions.java

```diff
@@ -0,0 +1,170 @@
+package com.mapbox.mapboxsdk.location;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.android.core.location.LocationEngineRequest;
+import com.mapbox.mapboxsdk.maps.Style;
+
+/**
+ * since 7.2.0
+ */
+public class LocationComponentActivationOptions {
+
+  private Context context;
+  private Style style;
+  private LocationEngine locationEngine;
+  private LocationEngineRequest locationEngineRequest;
+  private LocationComponentOptions locationComponentOptions;
+  private int styleRes;
+  private boolean useDefaultLocationEngine;
+
+  LocationComponentActivationOptions(@NonNull Context context, @NonNull Style style,
+                                     @Nullable LocationEngine locationEngine,
+                                     @Nullable LocationEngineRequest locationEngineRequest,
+                                     @Nullable LocationComponentOptions locationComponentOptions,
+                                     int styleRes,
+                                     boolean useDefaultLocationEngine) {
+    this.context = context;
+    this.style = style;
+    this.locationEngine = locationEngine;
+    this.locationEngineRequest = locationEngineRequest;
+    this.locationComponentOptions = locationComponentOptions;
+    this.styleRes = styleRes;
+    this.useDefaultLocationEngine = useDefaultLocationEngine;
+  }
+
+  /**
+   * Convenience method to retrieve a {@link LocationComponentActivationOptions} object which is ready to build with
+   *
+   * @return a builder object
+   */
+  @NonNull
+  public static Builder builder(@NonNull Context context, @NonNull Style fullyLoadedMapStyle) {
+    return new LocationComponentActivationOptions.Builder(context, fullyLoadedMapStyle);
+  }
+
+  /**
+   * The application's current context
+   *
+   * @return the application's current context
+   */
+  @Nullable
```

Comment:
Because the field should be final and the `Builder` will require `context`, this getter can be annotated as `NonNull`.

Reply:
Fixed